tests/func/test_data_cloud.py

```diff
@@ -229,9 +247,11 @@ def _setup_cloud(self):
         self._ensure_should_run()
 
         repo = self._get_url()
+        keyfile = self._get_keyfile()
 
         config = copy.deepcopy(TEST_CONFIG)
         config[TEST_SECTION][Config.SECTION_REMOTE_URL] = repo
+        config[TEST_SECTION][Config.SECTION_REMOTE_KEY_FILE] = keyfile
```

Comment:
Btw, you are mocking the connection itself, so keyfile, port, user, etc options passed through here don't even have any effect. Should we only start the ssh server and not mock the connection itself, so that RemoteSSH establishes the connection with the server based on these options by itself?

Reply:
Indeed. Actually, this was my initial proposal to not mock connection itself and left server open for further adjust. 
I will update this point.

Reply:
DONE